### PR TITLE
feat: added extended Euclidean algorithm

### PR DESCRIPTION
**Pull request by dsmurrow**
#### Description of Change

euclidean_algorithm_extended.c is implemented and lives in the math subdirectory.

In addition to finding the greatest common divisor of integers a and b, this algorithm will find values x and y such that $ax+by=gcd(a,b)$.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: added extended Euclidean algorithm

### CODE REVIEW

File: math/euclidean_algorithm_extended.c [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+static inline void xy_push(int arr[2], int newval)
+{
+    arr[1] = arr[0];
+    arr[0] = newval;
+}
+
+/**
+ * @brief calculates the value of x or y and push those into the small 'queues'
+ *
+ * Both x and y are found by taking their value from 2 iterations ago minus the
```
Comment (Panquesito7) [inline math/euclidean_algorithm_extended.c:45]:
```suggestion
 * @details Both x and y are found by taking their value from 2 iterations ago minus the
```

File: math/euclidean_algorithm_extended.c [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+ */
+static inline void calculate_next_xy(int quotient, int prev[2])
+{
+    int next = prev[1] - (prev[0] * quotient);
+    xy_push(prev, next);
+}
+
+/**
+ * @brief performs the extended Euclidean algorithm on integer inputs a and b
+ *
```
Comment (Panquesito7) [inline math/euclidean_algorithm_extended.c:62]: Missing parameter documentation.

File: math/euclidean_algorithm_extended.c [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    return result;
+}
+
+/** @} */
+
+/**
+ * @brief perform one single check on the result of the algorithm with provided
+ * parameters and expected output
+ *
```
Comment (Panquesito7) [inline math/euclidean_algorithm_extended.c:109]: No parameter documentation.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Review by Panquesito7 — APPROVED** [review]
Great work! Thanks for contributing. 🚀

**Review by tjgurwara99 — APPROVED** [review]

**Pull request merged**